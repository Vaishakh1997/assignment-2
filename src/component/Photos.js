import React, { Component } from 'react';
import { fetchPhotos } from '../Fetch-API/api-fetch';
import { connect } from 'react-redux'
import Carousel from './Carousel';

class Photos extends Component {
    componentDidMount() {
        const { albumId } = this.props.match.params
        this.props.photosList(albumId)
    }

    render() {
        const { loading, photos, error } = this.props

        return loading === true ?
            <div>Loading...</div> : <Carousel images={photos}/>
    }
}

const mapStateToProps = (state) => {
    return {
        loading: state.photosReducer.loading,
        photos: state.photosReducer.photos,
        error: state.photosReducer.error,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        photosList: (albumId) => dispatch(fetchPhotos(albumId)),
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Photos)

