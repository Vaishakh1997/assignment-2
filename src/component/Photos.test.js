import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { Photos } from './Photos'

const mockStore = configureMockStore();
const props= {loading: false, photos:[1,2], error:'error'}
const store = mockStore({});

let wrapped = shallow(
    <Provider store={store}>
        <Photos {...props}/>
    </Provider>
)


describe("Albums Component", () => {

    it("Albums should render without throwing an error", () => {
        expect(wrapped.debug()).toMatchSnapshot();
    });
    
    it('Expected props loading as false', () => {
        expect(wrapped.dive().props().loading).toEqual(props.loading)
    });

    it('Expected props albums as [1,2]', () => {
        expect(wrapped.dive().props().photos).toEqual(props.photos)
    });

    it('Expected props error as error message', () => {
        expect(wrapped.dive().props().error).toEqual(props.error)
    });
    
});