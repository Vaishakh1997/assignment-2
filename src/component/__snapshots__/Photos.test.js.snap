// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Albums Component Albums should render without throwing an error 1`] = `
"<ContextProvider value={{...}}>
  <Photos loading={false} photos={{...}} error=\\"error\\" />
</ContextProvider>"
`;
